payee/payeeApi_test.go

```diff
@@ -0,0 +1,105 @@
+package payee
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	_ "github.com/lib/pq"
+)
+
+var store PayeeRepository
+
+func initStore() PayeeRepository {
+	if store != nil {
+		return store
+	}
+	dsn := os.Getenv("TEST_DATABASE_URL")
+
+	db, err := sql.Open("postgres", dsn)
+	if err != nil {
+		panic(err)
+	}
+	store = PayeeDB(db)
+	return store
+}
+
+func cleanDB(db *sql.DB) error {
+	_, err := db.Exec("TRUNCATE payees RESTART IDENTITY CASCADE")
+	return err
+}
+
+func setupMux(t *testing.T) *http.ServeMux {
+	store := initStore()
+
+	payeeDb, ok := store.(*payeeDB)
+	if !ok {
+		t.Fatalf("store is not *payeeDB")
+	}
+
+	if err := cleanDB(payeeDb.db); err != nil {
+		t.Fatalf("failed to clean DB: %v", err)
+	}
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/payees", PayeePostAPI)
+
+	return mux
+}
+
+func TestPayeePostAPISuccess(t *testing.T) {
+	mux := setupMux(t)
+
+	payload := map[string]interface{}{
+		"name":           "Abdc",
+		"code":           "1262",
+		"account_number": 1234767893,
+		"ifsc":           "CBIN0123456",
+		"bank":           "CBI",
+		"email":          "abcd@example.com",
+		"mobile":         9876543292,
+		"category":       "Employee",
+	}
+	body, _ := json.Marshal(payload)
+
+	req := httptest.NewRequest(http.MethodPost, "/payees", bytes.NewBuffer(body))
+	w := httptest.NewRecorder()
+
+	mux.ServeHTTP(w, req)
+
+	type Response struct {
+		ID int `json:"id"`
+	}
+
+	var resp Response
+	err := json.Unmarshal([]byte(w.Body.Bytes()), &resp)
+	if err != nil {
+		t.Fatal("Error unmarshaling JSON:", err)
+		return
+	}
+
+	if w.Code != http.StatusCreated {
+		t.Fatalf("expected status %d, got %d, body=%s", http.StatusCreated, w.Code, w.Body.String())
+	}
+
+	if resp.ID != 1 {
+		t.Fatalf("The response body should be {\"id\":1}")
+	}
+}
+
+func TestPayeePostAPIInvalidJSON(t *testing.T) {
+	mux := setupMux(t)
+
+	req := httptest.NewRequest(http.MethodPost, "/payees", bytes.NewBufferString("{bad json}"))
```

Comment:
@Swarathmica-infraspec  
we are doing the route mapping in the test code itself and then testing it, it does not actually test the route mapping done in main.go